src/basicio.cpp

```diff
@@ -80,6 +80,7 @@ namespace Exiv2 {
     public:
         //! Constructor
         explicit Impl(std::string path);
+        ~Impl() = default;
```

Comment:
Why is it needed to add these default destructors?

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-special-member-functions.html is the check.

They're generated anyway, but it's good to have them explicit. Note the Options in that page. I have all of them enabled locally to avoid having to define all member functions.

Reply:
The rule C.21 says "If you define or =delete any copy, move, or destructor function, define or =delete them all" but in this case I think we should apply C.20: "If you can avoid defining default operations, do". 

In this specific case, there is only 1 explicit constructor. There are no copy or move constructors. I think it makes more sense to not define the constructor.

Reply:
Right but the presence of an explicit constructor violates the rule of zero, hence why this check is thrown.

Reply:
I have a little preference for not having these new lines of code, but it is just a minor thing. Everything else looks fine 👍